frontend/packages/kubevirt-plugin/src/components/create-vm-wizard/tabs/vm-settings-tab/vm-settings-tab.tsx

```diff
@@ -168,31 +112,19 @@ export class VMSettingsTabComponent extends React.Component<VMSettingsTabCompone
           cnvBaseImages={cnvBaseImages}
           onChange={this.props.onFieldChange}
         />
-        <FormFieldMemoRow
-          field={this.getField(VMSettingsField.PROVISION_SOURCE_TYPE)}
-          fieldType={FormFieldType.SELECT}
-        >
-          <FormField>
-            <FormSelect onChange={this.onChange(VMSettingsField.PROVISION_SOURCE_TYPE)}>
-              <FormSelectPlaceholderOption
-                placeholder={getPlaceholder(VMSettingsField.PROVISION_SOURCE_TYPE)}
-                isDisabled={!!this.getFieldValue(VMSettingsField.PROVISION_SOURCE_TYPE)}
-              />
-              {(this.getFieldAttribute(VMSettingsField.PROVISION_SOURCE_TYPE, 'sources') || []).map(
-                (source) => (
-                  <FormSelectOption key={source} value={source} label={source} />
-                ),
-              )}
-            </FormSelect>
-          </FormField>
-          {[
-            ProvisionSource.URL.toString(),
-            ProvisionSource.CONTAINER.toString(),
-            ProvisionSource.DISK.toString(),
-          ].includes(provisionSourceValue) && provisionSourceDiskHelpMsg}
-          {[ProvisionSource.PXE.toString()].includes(provisionSourceValue) &&
-            provisionSourceNetHelpMsg}
-        </FormFieldMemoRow>
+        <ProvisionSourceComponent
+          provisionSourceField={this.getField(VMSettingsField.PROVISION_SOURCE_TYPE)}
+          onChange={this.props.onFieldChange}
+          goToStorageStep={
+            steps[VMWizardTab.STORAGE]?.canJumpTo ? () => goToStep(VMWizardTab.STORAGE) : null
```

Comment:
can we use useCallback so the React.memo can kick in?

Reply:
it's a class component can't use hooks here

Reply:
let's convert it to a functional component then, it should be pretty painless.

Reply:
:exploding_head: it's a function now ...